### PR TITLE
Herzlich willkommen bei holiday check

### PR DESCRIPTION
**Pull request by konarkmodi**

### CODE REVIEW

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+Now we wanted to apply this belief to our call recordings and turn them into valuable lessons - It's like turning lead into gold, where every call transcript becomes a chance to make our customer service and product shine brighter.
+
+Several times it felt like walking through a maze, figuring things out step by step. A big moment for us was when we were able to repurpose an [Apple MacBook](https://en.wikipedia.org/wiki/Apple_M3) in a new way, thanks to its strong GPU.
+
+This wasn't just about saving money; it was a clever move that let us venture into running [LLMs](https://en.wikipedia.org/wiki/Large_language_model) and [ASR Models](https://huggingface.co/tasks/automatic-speech-recognition) locally - right where we work, keeping our customers' information safe and sound without compromising the efficiency of the results.
+
+***Mixing the Potion: Safe, Private, and Full of Surprises***
+
+We care a lot about keeping things safe and private, which means we use a lot of tools are built on Open Source and can be self-hosted with a license for commercial use, remixing them with our own special tricks.
```
Comment (tmayer) [inline _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md:38]: tools that are built on Open Source

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+<figure>
+  <audio controls src="img/posts/2024-03-11-herzlich-willkommen-bei-holiday-check/call.wav"></audio>
+</figure>
+
+**Step 1:** We use [Genesys Cloud](https://www.genesys.com/) as our Contact Center solution.
+Calls to which customers have consented to be recorded are periodically pulled from the Genesys API and go through an Audio transcribing and diarization pipeline based on [Pyannote audio](https://huggingface.co/pyannote/speaker-diarization) + [whisper-large-v3](https://openai.com/research/whisper).
+
+We also submit a [prompt to Whisper](https://cookbook.openai.com/examples/whisper_prompting_guide) with each chunk that goes into transcription process, this helps us feed our business vocabulary in order to further enhance the quality of the transcription.
+
+*The above call will transcribed as following:*
```
Comment (tmayer) [inline _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md:58]: will be transcribed as follows

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+**Step 2:** Even though we use self-hosted LLMs, but since raw transcripts can contain some form of personal information, we always do multiple scans locally using [Presidio](https://github.com/microsoft/presidio) and some custom prompts with Mistral. This really helps us remove majority of the personal information without loosing bits of conversation.
```
Comment (tmayer) [inline _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md:79]: remove the majority of the personal information without losing too many bits of the conversion

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+<figure>
+    <img src="img/posts/2024-03-11-herzlich-willkommen-bei-holiday-check/presidio.png" alt="diagram" class="centered" />
```
Comment (tmayer) [inline _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md:82]: did you also commit the presidio image? I can't find it in the list of files

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+5. The discounts apply to both travel and car rentals.
+6. The size of the voucher depends on the total cost of the rental; there are four tiers based on €100, €300, €500, and €1000 Euro rental costs.
+7. The customer can learn more about these benefits by visiting the HolidayCheck website or subscribing to their newsletter.
+8. The membership lasts for one year but can be cancelled before renewal with a simple email notification to HolidayCheck.
+9. Customers pay for the membership online and receive an invoice, which they can settle through bank transfer.
+10. They may join at any time during the year, and the benefits apply from that point onwards, even if they book their first trip before joining.
+11. The agent confirmed that customers do not need to provide bank information for membership sign-up or voucher redemption. Instead, they can pay online and receive an invoice for settlement.
+12. Customers can join the Premium Club at any time, and the benefits are valid indefinitely from the date of joining, with no specific time restrictions.
+Even long talks can be understood in just a little over a minute. That's pretty fast!*
```
Comment (tmayer) [inline _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md:110]: I would also add a section about evaluating the outcome and how we keep track that the summary/translation is good enough

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+***Finding Treasures: New Insights and Happy Customers***
+
+The results have been nothing short of remarkable. We've discovered intriguing correlations, such as the link between website features (or lack thereof) and customer dissatisfaction, allowing us to address issues proactively.
+The ultimate testament to the transformative power of AI will be when our agents are better equipped and our customers are happier - We are constantly striving to reach this goal.
+
+***Looking Ahead: New Discoveries Await***
+
+Using AI has really changed the game for us in customer service. We've found new ways to make things better, work more efficiently, and make our customers' experiences great. And there's more to come!
+
+We're excited about what the future holds, with lots of new ideas and AI projects on the horizon. Our journey into using technology to improve customer service is just getting started, and we can't wait to see where it takes us. Join us as we continue to mix technology with a personal touch, aiming for a future where every customer feels like they're on an epic adventure with us.
```
Comment (tmayer) [inline _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md:126]: do we want to mention that we explore such things in-house but then also investigate if there's a solution for that already existing and outsource it?

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md (hunk 5 above)
Comment (konarkmodi) [reply _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md:111]: Addes as "step 4"

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md (hunk 4 above)
Comment (konarkmodi) [reply _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md:83]: Eagle eyes. Added now.

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md (hunk 2 above)
Comment (konarkmodi) [reply _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md]: Done

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md (hunk 1 above)
Comment (konarkmodi) [reply _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md]: Done

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+We used some of the newest AI magic in 2023 to make our own special tool that helps us listen and learn from our calls.
+
+As we wrote at length before, we believe in the [power of Large Language Models (LLMs)](https://techblog.holidaycheck.com/post/2023/08/24/sentiment-analysis-llms), specially when it comes to their ability to understand human like text.
+Now we wanted to apply this belief to our call recordings and turn them into valuable lessons - It's like turning lead into gold, where every call transcript becomes a chance to make our customer service and product shine brighter.
+
+Several times it felt like walking through a maze, figuring things out step by step. A big moment for us was when we were able to repurpose an [Apple MacBook](https://en.wikipedia.org/wiki/Apple_M3) in a new way, thanks to its strong GPU.
+
+This wasn't just about saving money; it was a clever move that let us venture into running [LLMs](https://en.wikipedia.org/wiki/Large_language_model) and [ASR Models](https://huggingface.co/tasks/automatic-speech-recognition) locally - right where we work, keeping our customers' information safe and sound without compromising the efficiency of the results.
```
[the remark this answers is not in the document]
Comment (konarkmodi) [reply _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md:35]: Done, added in the title and intro paragraph too.

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+Let's dive into our story!
+
+***Our Adventure: From Rough Drafts to Winning Strategies***
+
+Think of all the valuable things we can learn from the hours of calls we have - ofcourse the ones which have consented ;)
+
+It's like having a treasure map but not knowing where X marks the spot. 
+We decided to face the big challenges of understanding our users and product better, at the same time saving money, and keeping our customers' talks private. 
+We used some of the newest AI magic in 2023 to make our own special tool that helps us listen and learn from our calls.
```
[the remark this answers is not in the document]
Comment (konarkmodi) [reply _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md]: Done. replaced it with advancements in the field of AI

File: _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+We're not just making waves; we're surfing the AI tide, harnessing the power of open-source models like [whisper-large-v3](https://openai.com/research/whisper) for understanding what people say and [Mistral](https://mistral.ai/) for helping us figure out how to use what they say to make things better. 
+
+Imagine a treasure trove of insights, locked within countless hours of call recordings. That's where we started our quest - Buckle up for an adventure where technology meets human touch, and discover how we're setting new benchmarks in user experience.
+
+Let's dive into our story!
+
+***Our Adventure: From Rough Drafts to Winning Strategies***
+
+Think of all the valuable things we can learn from the hours of calls we have - ofcourse the ones which have consented ;)
```
[the remark this answers is not in the document]
Comment (konarkmodi) [reply _posts/2024-03-11-herzlich-willkommen-bei-holiday-check.md]: done

**Review by tmayer — APPROVED** [review]
LGTM Nice blog post!

**Comment by konarkmodi** [comment]
thanks @tmayer . Waiting for some internal reviews, will merge once I get a go ahead.

**Pull request merged**